pyro/infer/mcmc/util.py

```diff
@@ -233,18 +233,47 @@ def _guess_max_plate_nesting(model, args, kwargs):
     return max_plate_nesting
 
 
-def _pe_maker(model, model_args, model_kwargs, trace_prob_evaluator, transforms):
-    def potential_energy(params):
-        params_constrained = {k: transforms[k].inv(v) for k, v in params.items()}
-        cond_model = poutine.condition(model, params_constrained)
-        model_trace = poutine.trace(cond_model).get_trace(*model_args, **model_kwargs)
-        log_joint = trace_prob_evaluator.log_prob(model_trace)
-        for name, t in transforms.items():
+class _PEMaker(object):
+    def __init__(self, model, model_args, model_kwargs, trace_prob_evaluator, transforms,
+                 jit_compile=False, jit_options=None, skip_jit_warnings=False):
+        self.model = model
+        self.model_args = model_args
+        self.model_kwargs = model_kwargs
+        self.trace_prob_evaluator = trace_prob_evaluator
+        self.transforms = transforms
+        self.jit_compile = jit_compile
+        self.jit_options = {"check_trace": False} if jit_options is None else jit_options
+        self.skip_jit_warnings = skip_jit_warnings
+        self._compiled_fn = None
+
+    def _potential_fn(self, params):
+        params_constrained = {k: self.transforms[k].inv(v) for k, v in params.items()}
+        cond_model = poutine.condition(self.model, params_constrained)
+        model_trace = poutine.trace(cond_model).get_trace(*self.model_args,
+                                                          **self.model_kwargs)
+        log_joint = self.trace_prob_evaluator.log_prob(model_trace)
+        for name, t in self.transforms.items():
             log_joint = log_joint - torch.sum(
                 t.log_abs_det_jacobian(params_constrained[name], params[name]))
         return -log_joint
 
-    return potential_energy
+    def _potential_fn_jit(self, params):
+        if self._compiled_fn:
+            return self._compiled_fn(params)
+        with pyro.validation_enabled(False), optional(ignore_jit_warnings(), self.skip_jit_warnings):
+            names, vals = zip(*sorted(params.items()))
+
+            def _pe_jit(*zi):
+                params = dict(zip(names, zi))
+                return self._potential_fn(params)
+
+            self._compiled_fn = torch.jit.trace(_pe_jit, vals, **self.jit_options)
+            return self._compiled_fn(*vals)
+
+    def get_potential_fn(self):
```

Comment:
nit: It seems a tad simpler to pass `jit_compile` and `jit_options` to this method, then below call
```py
pe_maker = _PEMaker(...)
init_params = _get_init_params(..., pe_maker.get_potential_fn(jit_compile=False))
potential_fn = pe_maker.get_potential_fn(jit_compile=jit_compile, jit_options=jit_options)
return init_params, potential_fn, transforms, model_trace
```